windivert/src/divert/mod.rs

```diff
@@ -343,6 +345,7 @@ impl<L: layer::WinDivertLayerTrait> WinDivert<L> {
 
     /// Handle close function.
     pub fn close(&mut self, action: CloseAction) -> Result<(), WinDivertError> {
```

Comment:
Forgot the changing `close` to be consuming

To avoid the issues in the `Drop` implementation and repeating code, maybe use a private inner close taking just `&self`

```suggestion
    pub fn close(self, action: CloseAction) -> Result<(), WinDivertError> {
        self.inner_close(action);
    }

    fn inner_close(&self, action: CloseAction) -> Result<(), WinDivertError> {
```